var/spack/repos/builtin/packages/bolt/package.py

```diff
@@ -40,3 +44,35 @@ def cmake_args(self):
         ]
 
         return options
+
+    @run_after('install')
+    def cache_test_sources(self):
+        """Copy the example source files after the package is installed to an
+        install test subdirectory for use during `spack test run`."""
+        self.cache_extra_test_sources(['examples'])
+
+    def run_sample_nested_example(self):
+        """Run stand alone test: sample_nested"""
+
+        test_dir = join_path(self.test_suite.current_test_cache_dir, 'examples')
+
+        if not os.path.exists(test_dir):
+            print('Skipping bolt test')
+            return
+
+        exe = 'sample_nested'
+
+        self.run_test('gcc',
```

Comment:
This worked for me as-is when I tested it, thanks!  

My concern in general is that the package is supposed to be buildable with a variety of compilers and this test case only checks one.  It is preferable that it use `self.compiler.cc` so the test builds with the same compiler used when the code is built.  Unfortunately, that simple change didn't work for me.  But the problem may be part of another issue.

Reply:
```suggestion
        # TODO: Either change to use self.compiler.cc (so using the build-time compiler)
        #  or add test parts that compile with the different supported compilers.
        self.run_test('gcc',
```